lib/Dancer2/Core/App.pm

```diff
@@ -896,10 +896,16 @@ sub template {
         $self->set_with_return( sub {
             $local_response ||= shift;
         });
-        my $content = $template->process( @_ );
+        # Catch any exceptions that may happen during template processing
+        my $content = eval { $template->process( @_ ) };
         $self->set_with_return($old_with_return);
+        # If there was a previous response set before the exception (or set as
+        # part of the exception handling), then use that, otherwise throw the
+        # exception as normal
         if ($local_response) {
             $self->with_return->($local_response);
+        } elsif ($@) {
+            die $@;
```

Comment:
I think it would be more robust to *immediately* take a copy of $@ after calling eval. For all you know, set_with_return - or any future code added between the eval and the check of $@ in the future - might modify $@, which is after all a global variable.

Reply:
Thanks, implemented with b36481f2

Reply:
Nice suggestion, @skington!